libfive/include/libfive/tree/opcode.hpp

```diff
@@ -79,6 +85,12 @@ namespace Opcode
     OPCODE(OP_ASIN, 13)     \
     OPCODE(OP_ACOS, 14)     \
     OPCODE(OP_ATAN, 15)     \
+    OPCODE(OP_SINH, 33)      \
```

Comment:
The point of `LIBFIVE_PACKED_OPCODES` is to make opcodes tightly ordered by number of arguments, e.g. so you can check whether an opcode has two arguments by seeing if it's `>= OP_ADD`.  This makes certain checks faster, but breaks compatibility with saved files (which is a tradeoff that [some folks](https://github.com/mkeeter/mpr/blob/master/CMakeLists.txt#L8) want to make).

Therefore, in this block, you should number these new starting at `16`, and bump the numbers of everything after them; the changes in the previous block are fine and correct.

(No worries about not figuring this out – it should probably be better commented)

Reply:
adding your text as a comment.